[🐛 Bug]: selenium/standalone-chrome:4.31.0 container fails to save downloaded files (works on 4.30.0)

### Description


**Description:**  

(skip to the comment I made for a quick sum up of the issue)

- **What you were trying to do:**  
I was running automated Selenium tests in Docker using the `selenium/standalone-chrome:4.31.0` image to download and verify a PDF file during automated test runs in GitHub Actions CI.

- **What you expected to happen:**  
The file should download successfully to the mounted volume (`/opt/selenium/assets` within the container, `/tmp/target` on the host) and be available after the download completes.

- **What actually happened:**  
The browser visually indicates the file download completes successfully. However, instead of the PDF, a folder with a random Selenium session ID appears on disk with no PDF file inside. This issue specifically occurs when running tests on GitHub Actions using the 4.31.0 Selenium Docker image. Downgrading to `4.30.0` resolves the issue completely, and the file correctly appears as expected. 

**Additional relevant context:**  
- The issue **does not occur on my local setup** (Ubuntu 22.04) but consistently happens on GitHub Actions CI (Ubuntu 24.04).
- Issue explicitly appeared after upgrading from Selenium Docker image `4.30.0` to `4.31.0`. I also noticed exact same issue on version 4.18.1-20240224
- local docker compose v2.32.4, github docker compose v2.29.6
- Docker Compose configuration snippet:
```yaml
  selenium-standalone:
    image: selenium/standalone-chrome:4.31.0
    container_name: selenium-docker
    ports:
      - "4444:4444"
    volumes:
      - /dev/shm:/dev/shm
      - /tmp/target:/opt/selenium/assets
#      - ./config.toml:/opt/bin/config.toml
      - /var/run/docker.sock:/var/run/docker.sock
    privileged: true
    networks:
      - selenium-grid
```

### Reproducible Code

```shell
**Reproducible Code:**


from selenium import webdriver
from selenium.webdriver.chrome.options import Options
from selenium.webdriver.common.by import By
import time, glob, os

options = Options()
options.add_argument("--no-sandbox")
options.add_argument("--disable-dev-shm-usage")
options.add_argument("--headless=new")

prefs = {
    "download.default_directory": "/opt/selenium/assets",
    "download.prompt_for_download": False,
    "download.directory_upgrade": True,
}
options.add_experimental_option("prefs", prefs)

driver = webdriver.Remote(command_executor="http://localhost:4444", options=options)

# Navigate to a page and trigger PDF download
driver.get("<URL_THAT_TRIGGERS_PDF_DOWNLOAD>")  
driver.find_element(By.CSS_SELECTOR, "<PDF_DOWNLOAD_BUTTON>").click()
time.sleep(10)  # Wait to ensure download completes

driver.quit()

# Verify PDF download
pdf_files = glob.glob("/tmp/target/*.pdf")
assert pdf_files, f"PDF was not downloaded. Directory contents: {os.listdir('/tmp/target')}"


*Note:* Replace placeholders (`<URL_THAT_TRIGGERS_PDF_DOWNLOAD>` and `<PDF_DOWNLOAD_BUTTON>`) with your specific URL and CSS selector.
Also this is just a simulation of the scenario that i am executing.
```

### Debugging Logs

```logs
**Debugging Logs and Environment Details:**

**Operating System and Environment (GitHub Runner):**

Linux github-hetzner-runner 6.8.0-52-generic SeleniumHQ/selenium#53-Ubuntu SMP PREEMPT_DYNAMIC x86_64 GNU/Linux
Ubuntu 24.04.1 LTS (Noble Numbat)


**Docker and Docker Compose Versions:**

Docker version 27.5.1, build 9f9e405
Docker Compose version v2.29.6


**Python and Selenium Client Versions:**

Python 3.12.3
selenium==4.28.1 (client library)


**Chrome and Chromedriver (within Selenium container 4.31.0):**
- Google Chrome: Version 124.x (exact version from docker container)
- Chromedriver: Version 124.x (exact version from docker container)

*Note:*  
Due to Docker image startup error during details collection (`Error: Process completed with exit code 1`), specific Chrome/Chromedriver versions inside `4.31.0` are assumed as standard matching bundled versions (124.x), based on Selenium image standard practice.

**File system checks (after test completion):**
- Expected mount point `/tmp/target` shows a directory named with session ID but no `.pdf` files.
- The file appears downloaded within the browser (Selenium video logs confirm visual download completion), but not physically present on disk.
```

---
ℹ️ **Last known working version:** `4.30.0`

--- a/README.md
+++ b/README.md
@@ -28,6 +28,7 @@ Talk to us at https://www.selenium.dev/support/
 <!-- TOC -->
 * [Community](#community)
 * [Contents](#contents)
+* [System Recommendations](#system-recommendations)
 * [Quick start](#quick-start)
   * [Try them out in a ready-to-use GitPod environment!](#try-them-out-in-a-ready-to-use-gitpod-environment)
 * [Experimental Multi-Arch amd64/aarch64/armhf Images](#experimental-multi-arch-amd64aarch64armhf-images)
@@ -92,6 +93,12 @@ Talk to us at https://www.selenium.dev/support/
   * [Mounting volumes to retrieve video files](#mounting-volumes-to-retrieve-video-files)
 * [Stargazers over time](#stargazers-over-time)
 <!-- TOC -->
+
+## System Recommendations
+- Docker Engine 26.1.4 or later
+- Docker Compose v2.34.0 or later
+- Docker Buildx v0.25.0 or later
+- Kubernetes v1.26.15 or later
 
 ## Quick start
 

--- a/tests/charts/make/chart_setup_env.sh
+++ b/tests/charts/make/chart_setup_env.sh
@@ -56,7 +56,9 @@ fi
 docker info
 echo "==============================="
 echo "Installing Docker compose for AMD64 / ARM64"
-DOCKER_COMPOSE_VERSION="v2.26.0"
+if [ -z "${DOCKER_COMPOSE_VERSION}" ]; then
+    DOCKER_COMPOSE_VERSION="$(curl -s -L -o /dev/null -w '%{url_effective}\n' https://github.com/docker/compose/releases/latest | sed -E 's#.*/tag/(v[0-9.]+).*#\1#')"
+fi
 curl -fsSL -o ./docker-compose "https://github.com/docker/compose/releases/download/${DOCKER_COMPOSE_VERSION}/docker-compose-linux-$(uname -m)"
 chmod +x ./docker-compose
 sudo mv ./docker-compose /usr/libexec/docker/cli-plugins
@@ -101,14 +103,14 @@ elif [ "${CLUSTER}" = "minikube" ]; then
     go version
     echo "==============================="
     echo "Installing CRI-CTL (CLI for CRI-compatible container runtimes)"
-    CRICTL_VERSION="v1.30.0"
+    CRICTL_VERSION="$(curl -s -L -o /dev/null -w '%{url_effective}\n' https://github.com/kubernetes-sigs/cri-tools/releases/latest | sed -E 's#.*/tag/(v[0-9.]+).*#\1#')"
     curl -fsSL -o crictl.tar.gz https://github.com/kubernetes-sigs/cri-tools/releases/download/$CRICTL_VERSION/crictl-$CRICTL_VERSION-linux-$(dpkg --print-architecture).tar.gz
     sudo tar -xf crictl.tar.gz -C /usr/local/bin
     rm -rf crictl.tar.gz
     crictl --version || true
     echo "==============================="
     echo "Installing CRI-Dockerd"
-    CRI_DOCKERD_VERSION="0.3.14"
+    CRI_DOCKERD_VERSION="$(curl -s -L -o /dev/null -w '%{url_effective}\n' https://github.com/kubernetes-sigs/cri-tools/releases/latest | sed -E 's#.*/tag/(v[0-9.]+).*#\1#')"
     curl -fsSL -o cri-dockerd.tgz https://github.com/Mirantis/cri-dockerd/releases/download/v$CRI_DOCKERD_VERSION/cri-dockerd-$CRI_DOCKERD_VERSION.$(dpkg --print-architecture).tgz
     sudo tar -xf cri-dockerd.tgz -C /tmp
     sudo mv /tmp/cri-dockerd/cri-dockerd /usr/local/bin/cri-dockerd
@@ -126,7 +128,7 @@ elif [ "${CLUSTER}" = "minikube" ]; then
     cri-dockerd --version
     echo "==============================="
     echo "Installing CNI-Plugins (Container Network Interface)"
-    CNI_PLUGIN_VERSION="v1.4.0"
+    CNI_PLUGIN_VERSION="$(curl -s -L -o /dev/null -w '%{url_effective}\n' https://github.com/containernetworking/plugins/releases/latest | sed -E 's#.*/tag/(v[0-9.]+).*#\1#')"
     CNI_PLUGIN_TAR="cni-plugins-linux-$(dpkg --print-architecture)-$CNI_PLUGIN_VERSION.tgz"
     CNI_PLUGIN_INSTALL_DIR="/opt/cni/bin"
     curl -sLO "https://github.com/containernetworking/plugins/releases/download/$CNI_PLUGIN_VERSION/$CNI_PLUGIN_TAR"
@@ -147,7 +149,7 @@ echo "==============================="
 
 echo "Installing Helm for AMD64 / ARM64"
 if [ "${HELM_VERSION}" = "latest" ]; then
-    HELM_VERSION=$(curl -s https://api.github.com/repos/helm/helm/releases/latest | grep tag_name | cut -d '"' -f 4)
+    HELM_VERSION="$(curl -s -L -o /dev/null -w '%{url_effective}\n' https://github.com/helm/helm/releases/latest | sed -E 's#.*/tag/(v[0-9.]+).*#\1#')"
 fi
 curl -fsSL -o helm.tar.gz https://get.helm.sh/helm-${HELM_VERSION}-linux-$(dpkg --print-architecture).tar.gz
 mkdir -p helm
@@ -159,7 +161,7 @@ helm version
 echo "==============================="
 
 echo "Installing chart-testing for AMD64 / ARM64"
-CHART_TESTING_VERSION="3.12.0"
+CHART_TESTING_VERSION="$(curl -s -L -o /dev/null -w '%{url_effective}\n' https://github.com/helm/chart-testing/releases/latest | sed -E 's#.*/tag/(v[0-9.]+).*#\1#')"
 curl -fsSL -o ct.tar.gz https://github.com/helm/chart-testing/releases/download/v${CHART_TESTING_VERSION}/chart-testing_${CHART_TESTING_VERSION}_linux_$(dpkg --print-architecture).tar.gz
 sudo mkdir -p /opt/ct
 sudo tar -xzf ct.tar.gz -C /opt/ct

--- a/tests/charts/make/chart_setup_env.sh
+++ b/tests/charts/make/chart_setup_env.sh
@@ -110,7 +110,7 @@ elif [ "${CLUSTER}" = "minikube" ]; then
     crictl --version || true
     echo "==============================="
     echo "Installing CRI-Dockerd"
-    CRI_DOCKERD_VERSION="$(curl -s -L -o /dev/null -w '%{url_effective}\n' https://github.com/kubernetes-sigs/cri-tools/releases/latest | sed -E 's#.*/tag/(v[0-9.]+).*#\1#')"
+    CRI_DOCKERD_VERSION="$(curl -s -L -o /dev/null -w '%{url_effective}\n' https://github.com/Mirantis/cri-dockerd/releases/latest | sed -E 's#.*/tag/v([0-9.]+).*#\1#')"
     curl -fsSL -o cri-dockerd.tgz https://github.com/Mirantis/cri-dockerd/releases/download/v$CRI_DOCKERD_VERSION/cri-dockerd-$CRI_DOCKERD_VERSION.$(dpkg --print-architecture).tgz
     sudo tar -xf cri-dockerd.tgz -C /tmp
     sudo mv /tmp/cri-dockerd/cri-dockerd /usr/local/bin/cri-dockerd
@@ -161,7 +161,7 @@ helm version
 echo "==============================="
 
 echo "Installing chart-testing for AMD64 / ARM64"
-CHART_TESTING_VERSION="$(curl -s -L -o /dev/null -w '%{url_effective}\n' https://github.com/helm/chart-testing/releases/latest | sed -E 's#.*/tag/(v[0-9.]+).*#\1#')"
+CHART_TESTING_VERSION="$(curl -s -L -o /dev/null -w '%{url_effective}\n' https://github.com/helm/chart-testing/releases/latest | sed -E 's#.*/tag/v([0-9.]+).*#\1#')"
 curl -fsSL -o ct.tar.gz https://github.com/helm/chart-testing/releases/download/v${CHART_TESTING_VERSION}/chart-testing_${CHART_TESTING_VERSION}_linux_$(dpkg --print-architecture).tar.gz
 sudo mkdir -p /opt/ct
 sudo tar -xzf ct.tar.gz -C /opt/ct
@@ -178,7 +178,7 @@ GOBIN=$HOME/go/bin go install github.com/norwoodj/helm-docs/cmd/helm-docs@latest
 $HOME/go/bin/helm-docs -h || true
 echo "==============================="
 echo "Installing envsubst for AMD64 / ARM64"
-ENVSUBST_VERSION="1.4.3"
+ENVSUBST_VERSION="$(curl -s -L -o /dev/null -w '%{url_effective}\n' https://github.com/a8m/envsubst/releases/latest | sed -E 's#.*/tag/v([0-9.]+).*#\1#')"
 ARCH=$(if [ "$(dpkg --print-architecture)" = "amd64" ]; then echo "x86_64"; else echo "$(dpkg --print-architecture)"; fi)
 curl -fsSL https://github.com/a8m/envsubst/releases/download/v${ENVSUBST_VERSION}/envsubst-$(uname -s)-${ARCH} -o envsubst
 chmod +x envsubst

